src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/NumberBox/NumberBoxTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.UI.Xaml.Controls;
+using MUXControlsTestApp.Utilities;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Automation;
+using MUXControlsTestApp;
+
+namespace Uno.UI.RuntimeTests.MUX.Microsoft_UI_Xaml_Controls
+{
+	[TestClass]
+	public class NumberBoxTests : MUXApiTestBase
+	{
+		//[TestMethod]
+		//public void VerifyTextAlignmentPropogates()
+		//{
+		//	var numberBox = SetupNumberBox();
+		//	TextBox textBox = null;
+
+		//	RunOnUIThread.Execute(() =>
+		//	{
+		//		Content.UpdateLayout();
+
+		//		textBox = TestUtilities.FindDescendents<TextBox>(numberBox).Where(e => e.Name == "InputBox").Single();
+		//		Assert.AreEqual(TextAlignment.Left, textBox.TextAlignment, "The default TextAlignment should be left.");
+
+		//		numberBox.TextAlignment = TextAlignment.Right;
+		//		Content.UpdateLayout();
+
+		//		Assert.AreEqual(TextAlignment.Right, textBox.TextAlignment, "The TextAlignment should have been updated to Right.");
+		//	});
+		//}
+
+		//[TestMethod]
+		//public void VerifyInputScopePropogates()
+		//{
+		//	var numberBox = SetupNumberBox();
+
+		//	RunOnUIThread.Execute(() =>
+		//	{
+		//		Content.UpdateLayout();
+		//		var inputTextBox = TestUtilities.FindDescendents<TextBox>(numberBox).Where(e => e.Name == "InputBox").Single();
+
+		//		Assert.AreEqual(1, inputTextBox.InputScope.Names.Count);
+		//		Assert.AreEqual(InputScopeNameValue.Number, inputTextBox.InputScope.Names[0].NameValue, "The default InputScope should be 'Number'.");
+
+		//		var scopeName = new InputScopeName();
+		//		scopeName.NameValue = InputScopeNameValue.CurrencyAmountAndSymbol;
+		//		var scope = new InputScope();
+		//		scope.Names.Add(scopeName);
+
+		//		numberBox.InputScope = scope;
+		//		Content.UpdateLayout();
+
+		//		Assert.AreEqual(1, inputTextBox.InputScope.Names.Count);
+		//		Assert.AreEqual(InputScopeNameValue.CurrencyAmountAndSymbol, inputTextBox.InputScope.Names[0].NameValue, "The InputScope should be 'CurrencyAmountAndSymbol'.");
+		//	});
+
+		//	return;
+		//}
```

Comment:
We don't have `InputScope` or `TextAlignment` properties.

I even can't find them in the API reference [here](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.numberbox?view=winui-3.0), but they're implemented in MUX repo.

https://github.com/microsoft/microsoft-ui-xaml/pull/3186
https://github.com/microsoft/microsoft-ui-xaml/pull/2901

I have no idea if we should port these two PRs and enable those commented tests.

Reply:
We should be updating the numberbox code to the latest, which is likely to include what's missing. It looks like the current implementation is using commit 2237b09d9ba6052414ddccf3d61316de20564303 from winui.

Reply:
@jeromelaban Attempted to do that in 2a95376. I'm hoping I did it correctly :)